### PR TITLE
[TESTMERGE ONLY] dumb hotfix

### PR DESCRIPTION
**Pull request by Putnam3145**
## About The Pull Request

Panic!

## Why It's Good For The Game

server stop crash please

## Changelog
:cl:
fix: Vendor icons are gone
/:cl:

### CODE REVIEW

**Review by Ghommie — APPROVED** [review]

File: code/modules/vending/_vending.dm [hunk 1]
```diff
@@ -369,9 +369,8 @@ IF YOU MODIFY THE PRODUCTS LIST OF A MACHINE, MAKE SURE TO UPDATE ITS RESUPPLY C
 /obj/machinery/vending/proc/GetIconForProduct(datum/data/vending_product/P)
 	if(vending_cache[P.product_path])
 		return vending_cache[P.product_path]
-	var/product = new P.product_path()
-	vending_cache[P.product_path] = icon2base64(getFlatIcon(product, no_anim = TRUE))
-	qdel(product)
+	var/product = P.product_path
```
Comment (Ghommie) [inline code/modules/vending/_vending.dm]:
```suggestion
	var/atom/product = P.product_path
```

**Review by Ghommie — COMMENTED** [review]
Umn, nope, there is stuff to do.

File: code/modules/vending/_vending.dm [hunk 2]
```diff
@@ -370,7 +370,11 @@ IF YOU MODIFY THE PRODUCTS LIST OF A MACHINE, MAKE SURE TO UPDATE ITS RESUPPLY C
 	if(vending_cache[P.product_path])
 		return vending_cache[P.product_path]
 	var/product = new P.product_path()
-	vending_cache[P.product_path] = icon2base64(getFlatIcon(product, no_anim = TRUE))
+	try
+		vending_cache[P.product_path] = icon2base64(getFlatIcon(product, no_anim = TRUE))
```
Comment (LetterN) [inline code/modules/vending/_vending.dm:374]: try `product.icon` maybe?

**Comment by Linzolle** [comment]
tgui next vendors were merged so this isn't needed anymore.

**Pull request closed without merging**